### PR TITLE
Rebased and updated patches from Neil's branch.

### PR DESCRIPTION
**Pull request by brett19**

### CODE REVIEW

File: expression_json.go [hunk 1]
```diff
@@ -31,71 +31,77 @@ func parseJsonField(data []interface{}) (Expression, error) {
 	return out, nil
 }
 
-func parseJsonEquals(data []interface{}) (Expression, error) {
+func parseJsonComparison(data []interface{}) (Expression, Expression, error) {
 	lhsData, ok := data[1].([]interface{})
 	if !ok {
-		return nil, errors.New("invalid equals expression lhs format")
+		return nil, nil, errors.New("invalid equals expression lhs format")
```
Comment (nelio2k) [inline expression_json.go]: this function is used by more than just equals (i.e. not equals). So this error may not reflect the accurate caller.

File: expression_json.go [hunk 2]
```diff
@@ -31,71 +31,77 @@ func parseJsonField(data []interface{}) (Expression, error) {
...
 	}
 
 	rhsData, ok := data[2].([]interface{})
 	if !ok {
-		return nil, errors.New("invalid equals expression rhs format")
+		return nil, nil, errors.New("invalid equals expression rhs format")
```
Comment (nelio2k) [inline expression_json.go]: this function is used by more than just equals (i.e. not equals). So this error may not reflect the accurate caller.

File: transform.go [hunk 3]
```diff
@@ -63,18 +63,27 @@ type OpType int
...
 func opTypeToString(value OpType) string {
 	switch value {
 	case OpTypeEquals:
 		return "eq"
+	case OpTypeNotEquals:
+		return "neq"
 	case OpTypeLessThan:
 		return "lt"
-	case OpTypeGreaterEquals:
+	case OpTypeLessEqual:
```
Comment (nelio2k) [inline transform.go]: nit-pick: Did you want to make it consistently "*Equals"?
i.e. LessEquals, NotEquals, GreaterEquals... whereas right now it's:
NotEqual**s**, LessEqual, GreaterEqual
It's fine as is, though.

Comment (brett19) [reply transform.go]: I actually struggled with what to do here.  This ticket however targets the specific task of adding additional expressions, not renaming old expressions.  I was trying to keep the commits atomic and independent.  I'll change the name of the existing ones in a commit before this one and push it up though.

File: expression_json.go (hunk 1 above)
Comment (brett19) [reply expression_json.go]: Good call.  I actually had to fix this exact problem in the core expression side earlier today, should have caught it!  Will fix it shortly.

File: transform.go (hunk 3 above)
Comment (brett19) [reply transform.go]: Done.

File: expression_json.go (hunk 1 above)
Comment (brett19) [reply expression_json.go]: Done.

**Pull request merged**